plugins/retry/src/retry.spec.ts

```diff
@@ -1,7 +1,283 @@
+
+    // initial fetch fails
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    // first retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+    // no second retry because the default retry is 1
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+  })
+
+  it('retry option with a number', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+
+    factory({
+      key: ['key'],
+      query,
+      retry: 2,
+    })
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    // first retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+    // second retry
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(3)
+    // no further retries
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(3)
+  })
+
+  it('retry option with a function', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+    // stop retry when the error message is "ko"
+    const retry: RetryOptions['retry'] = (attempt, error) => {
+      if ((error as Error).message === 'ko') return false
+      return attempt < 2
+    }
+
+    factory({
+      key: ['key'],
+      query,
+      retry,
+    })
+    await flushPromises()
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+  })
+
+  it('delay option with a number', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+    const delay = 500
+
+    factory({
+      key: ['key'],
+      query,
+      retry: { delay },
+    })
+
+    // retry occurs exactly after 500ms
+    await flushPromises()
+    vi.advanceTimersByTime(RETRY_OPTIONS_DEFAULTS.delay - 100)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    vi.advanceTimersByTime(100)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+  })
+
+  it('delay option with a function', async () => {
+    const query = vi.fn(async () => {
+      throw new Error('ko')
+    })
+    // 500ms for first retry, 1500ms for subsequent retries
+    const delay: RetryOptions['delay'] = (attempt) => {
+      return attempt === 0 ? 500 : 1500
+    }
+
+    factory({
+      key: ['key'],
+      query,
+      retry: { retry: 2, delay },
+    })
+    //  initial fetch fails
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(1)
+    // first retry after 500ms
+    vi.advanceTimersByTime(500)
+    await flushPromises()
+    expect(query).toHaveBeenCalledTimes(2)
+    // second retry after 1500ms
+    vi.advanceTimersByTime(1500)
```

Comment:
@coderabbitai let's split this timer advance in two: let's advance 1000 first, check that the query was still called 2 times, then advance 500 to reach 1500 and these that the query is called 3 times

Reply:
use a `suggestion` tag instead of `diff` so that I can apply the change

Reply:
fix the indentation

Reply:
```suggestion
    // advance 1000ms (not enough time for next retry)
    vi.advanceTimersByTime(1000)
    await flushPromises()
    expect(query).toHaveBeenCalledTimes(2)
    // advance remaining 500ms to trigger second retry
    vi.advanceTimersByTime(500)
    await flushPromises()
    expect(query).toHaveBeenCalledTimes(3)
```